homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -411,16 +629,21 @@ async def async_stop_cover_tilt(self, **kwargs):
     @callback
     def _async_process_updated_position_data(self, position_data):
         """Process position data."""
-        if ATTR_POSKIND1 not in position_data:
-            return
-        if int(position_data[ATTR_POSKIND1]) == POSKIND_PRIMARY:
-            self._current_hd_cover_position = int(position_data[ATTR_POSITION1])
-            self._attr_current_cover_tilt_position = 0
-        if int(position_data[ATTR_POSKIND1]) == POSKIND_VANE:
-            self._current_hd_cover_position = MIN_POSITION
-            self._attr_current_cover_tilt_position = hd_position_to_hass(
-                int(position_data[ATTR_POSITION1]), self._max_tilt
-            )
+        if ATTR_POSITION1 in position_data:
+            if int(position_data[ATTR_POSKIND1]) == POS_KIND_PRIMARY:
+                self.set_position_primary(int(position_data[ATTR_POSITION1]))
+                self._current_hd_cover_vane = MIN_POSITION
+            if int(position_data[ATTR_POSKIND2]) == POS_KIND_VANE:
+                self.set_position_primary(MIN_POSITION)
+                self._current_hd_cover_vane = hd_position_to_hass(
+                    int(position_data[ATTR_POSITION1]), self._max_tilt
+                )
```

Comment:
Let's abstract `_current_hd_cover_vane` away here as well like you did with `set_position_primary`

Reply:
I didn't see a change for this. I think its the only thing left

Reply:
Maybe `set_position_vane` is a good name here